callback-examples/callback-example.jl

```diff
@@ -0,0 +1,36 @@
+using JSOSolvers, LaTeXStrings, Logging, NLPModels, Plots, ADNLPModels
```

Comment:
I am wondering if this should move to the docs instead of a separate folder. What do you think?

Reply:
Yes, I agree.

Reply:
I moved it, but it is also a little bit different.

Reply:
Thanks!